src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +53,176 @@ class MissingProjectMembership extends React.Component<Props, State> {
-    if (features.has('open-membership')) {
-      return t('To view this data you must one of the following teams.');
-    } else {
-      return t(
-        'To view this data you must first request access to one of the following teams:'
-      );
-    }
-  }
-
-  renderJoinTeams(features: Set<string>) {
+  getTeamsForAccess() {
+    const request: string[] = [];
+    const pending: string[] = [];
     const teams = this.state.project?.teams ?? [];
+    const team = teams.map(tm => TeamStore.getBySlug(tm.slug));
+    team.map(tm => (tm?.isPending ? pending.push(tm!.slug) : request.push(tm!.slug)));
+    return [request, pending];
+  }
 
-    if (!teams.length) {
-      return (
-        <EmptyMessage>
-          {t(
-            'No teams have access to this project yet. Ask an admin to add your team to this project.'
-          )}
-        </EmptyMessage>
-      );
-    }
+  handleChange = (evt: string) => {
+    const input = evt;
+    this.setState({team: input});
+  };
 
-    return teams.map((team: Team) => (
-      <p key={team.slug}>
-        #{team.slug}: {this.renderJoinTeam(team, features)}
-      </p>
-    ));
-  }
+  renderPendingTeam = (team: string) => {
+    return {
+      value: team,
+      label: <DisabledLabel>{`#${team}`}</DisabledLabel>,
+    };
+  };
 
   render() {
-    const {organization} = this.props;
+    const {organization, groupId} = this.props;
+    const team = this.state.team;
+    const teams = this.state.project?.teams ?? [];
     const features = new Set(organization.features);
 
+    const teamAccess = [
+      {
+        label: t('Request Access'),
+        options: this.getTeamsForAccess()[0].map(request => ({
+          value: request,
+          label: `#${request}`,
+        })),
+      },
+      {
+        label: t('Pending Requests'),
+        options: this.getTeamsForAccess()[1].map(pending =>
+          this.renderPendingTeam(pending)
+        ),
+      },
+    ];
+
     return (
       <div className="container">
-        <StyledWell centered>
-          <StyledIconFlag size="xxl" />
-          <p>{t("You're not a member of this project.")}</p>
-          <p>{this.renderExplanation(features)}</p>
-          {this.renderJoinTeams(features)}
-        </StyledWell>
+        <Panel>
+          {!teams.length ? (
+            <EmptyMessage icon={<StyledIconFlag size="xl" color="gray300" />}>
+              {t(
+                'No teams have access to this project yet. Ask an admin to add your team to this project.'
+              )}
+            </EmptyMessage>
+          ) : (
+            <EmptyMessage
+              icon={<StyledIconFlag size="xl" color="gray300" />}
+              title={t("You're not a member of this project.")}
+              description={t(
+                `You'll need to join a team with access to Issue ID ${groupId} before you can view it.`
+              )}
+              action={
+                <StyledField>
+                  <StyledSelectField
```

Comment:
So now that you removed the form, I think you can use a `SelectControl` here instead of `SelectField`. The `-Field` version of these form elements are used (generally) in combination with a `<Form>` and it includes a label with an optional help/description text. Since we don't use these, we can use the `SelectControl` itself.